runtime/lua/vim/provider/node.lua

```diff
@@ -0,0 +1,23 @@
+local M = {}
+
+--- @param version string?
```

Comment:
```suggestion
--- @param version? string
```

Reply:
Unsure why this syntax is more preferable tbh, but why not

Reply:
Hmm I think `version string?` is correct, because `min_version` is not an optional parameter. This is a local function, so shouldn't be omitted as an optional param.

Reply:
Yeah I dunno if it really matters tbh. I can revert it

Reply:
There's no meaningful difference between the two. 

That being said, prefer the same convention as "nearby" code (which is not helpful here since you're writing this whole provider code from scratch.)

So if in doubt, use what Lewis' recent PR made docgen do (which I think is name? type, but please double check.)

Reply:
I'd like avoid hunting for a convention that may or may not exist for such an inconsequential thing. @lewis6991 do you have a preference for how that above should look like, if so we can go with that.

Reply:
That was my point. The only real convention is the vimdoc script, which cannot but be consistent. I don't want to enforce this, though, without need. Otherwise, just avoid mixing conventions in a single file (which I thought went without saying).

But I'll happily check the PR for (from which this is obvious).

Reply:
Bloody hell, fine I'll take a look. It looks like he used `name boolean?` convention, so let's go with that.

Reply:
I agree with wookayin. Using `?` on the name should indicate to **the reader** that this argument doesn't need to be supplied rather than just being allowed to be `nil`. This can be detected in Lua using `select('#', ...)`, meaning `foo() !== foo(nil)`

But when I doubt just always tag `?` onto the type.